lbrynet/lbrynet_daemon/auth/server.py

```diff
@@ -125,36 +143,35 @@ def expire_session():
 
         try:
             self._run_subhandlers(request)
-        except SubhandlerError:
-            return server.failure
+        except SubhandlerError as err:
+            self._render_error(request, err, version)
+            return server.NOT_DONE_YET
 
         reply_with_next_secret = False
         if self._use_authentication:
             if function_name in self.authorized_functions:
                 try:
                     self._verify_token(session_id, parsed, token)
-                except InvalidAuthenticationToken:
-                    log.warning("API validation failed")
-                    request.setResponseCode(self.UNAUTHORIZED)
-                    request.finish()
+                except InvalidAuthenticationToken as err:
+                    log.error("API validation failed")
+                    self._render_error(request, err, version, response_code=AuthJSONRPCServer.UNAUTHORIZED)
                     return server.NOT_DONE_YET
                 self._update_session_secret(session_id)
                 reply_with_next_secret = True
 
         try:
             function = self._get_jsonrpc_method(function_name)
-        except Exception:
-            log.warning("Unknown method: %s", function_name)
-            return server.failure
+        except AttributeError as err:
+            log.error("Unknown method: %s", function_name)
```

Comment:
is an error here the fault of the caller?  if so, probably want to change this to a warning